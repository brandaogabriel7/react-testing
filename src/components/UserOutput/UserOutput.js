import React from 'react';

const UserOutput = ({ username }) => {
    return (
        <div>
            <p>Username: {username}</p>
            <p>I hope I'll be overwritten!</p>
        </div>
    );
};

export default UserOutput;