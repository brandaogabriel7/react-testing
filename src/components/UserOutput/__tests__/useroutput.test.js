import { render, screen } from '@testing-library/react';

import UserOutput from '../UserOutput';

const renderUsernameCorrectlyTestCases = [
    ['myUsername123', 'Username: myUsername123', 'I hope I\'ll be overwritten!'],
    ['brandsons', 'Username: brandsons', 'I hope I\'ll be overwritten!'],
    ['theBest001', 'Username: theBest001', 'I hope I\'ll be overwritten!']
];

const matchSnapshotTestCases = [
    ['randomUsername123'],
    ['anotherUsename']
];

describe('user output should', () => {
    it.each(renderUsernameCorrectlyTestCases)('render \'%s\' username correctly', (username, firstP, secondP) => {
        render(<UserOutput username={username} />);
        expect(screen.getByText(firstP)).toBeInTheDocument();
        expect(screen.getByText(secondP)).toBeInTheDocument();
    });

    it.each(matchSnapshotTestCases)('match snapshot for \'%s\' username', (username) => {
        const { container } = render(<UserOutput username={username} />);
        expect(container).toMatchSnapshot();
    });
});