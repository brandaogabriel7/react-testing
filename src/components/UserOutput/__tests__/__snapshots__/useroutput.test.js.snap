// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user output should match snapshot for 'anotherUsename' username 1`] = `
<div>
  <div>
    <p>
      Username: 
      anotherUsename
    </p>
    <p>
      I hope I'll be overwritten!
    </p>
  </div>
</div>
`;

exports[`user output should match snapshot for 'randomUsername123' username 1`] = `
<div>
  <div>
    <p>
      Username: 
      randomUsername123
    </p>
    <p>
      I hope I'll be overwritten!
    </p>
  </div>
</div>
`;
